### PR TITLE
Register login Validation

### LINKED ISSUE #218: [FEATURE] Real-time validation in login and Register form
## Description
Currently, for the register and login form, there is no real-time validation where the user gets a validation message as the user enters the input value. 

## Solution
I want to implement the functionality to display the validation messages below respective fields as the user enters the value and the validation message will disappear as soon as the valid input is given.
This will implement from sending invalid data to the backend it also enhances processing speed in the backend.
I will write a reusable code so that it can be implemented in other forms as well like.

### Screenshots

I want to implement functionality similar to below

## Sample Video

https://github.com/agamjotsingh18/pollitup/assets/86224794/b3d8c27f-da3b-4b82-9ea2-e4db3c710923

## Request
If you find this issue worth implementing please assign this issue to me under gssoc label as I am a gssoc contributor

### PR DESCRIPTION
**Pull request by AlfiyaSiddique**
# Pull Request Template

## Description
Added Inline Validation in register and login forms.

Fixes #218 **[Feature] Real-time validation in login and Register form**

## Type of change

- [x] New feature (non-breaking change which adds functionality)
#Sample Output

https://github.com/agamjotsingh18/pollitup/assets/86224794/2ffe2f9a-49e8-480a-b353-a8366f54991b

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/assets/validation.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
+const emailRegex = /^\w+(?:[\.-]\w+)*@(?:gmail\.com|yahoo\.com|hotmail\.com|aol\.com|outlook\.com)$/;
```
Comment (Arnabdaz) [inline src/assets/validation.js:1]: image
you need to update the email regex currently it only accepts a small number of domain ... remove the requirement for a specific domain and let users use anything separated with dot.

Comment (AlfiyaSiddique) [reply src/assets/validation.js:1]: okay got it. 👍🏻

**Comment by AlfiyaSiddique** [comment]
Hello @Arnabdaz 

The above requested changes are implemented.

**Sample Output**

https://github.com/agamjotsingh18/pollitup/assets/86224794/4c4f8155-d13a-4a14-987a-f24f8ad4d13b

**Comment by AlfiyaSiddique** [comment]
Hey @Arnabdaz , 

The requested changes are implemented. Please review this PR!

**Review by Arnabdaz — APPROVED** [review]

**Pull request merged**